llama_index/command_line/rag.py

```diff
@@ -0,0 +1,225 @@
+                "summarizer": response_synthesizer,
+            }
+        )
+        query_pipeline.add_link("query", "retriever")
+        query_pipeline.add_link("retriever", "summarizer", dest_key="nodes")
+        query_pipeline.add_link("query", "summarizer", dest_key="query_str")
+        return query_pipeline
+
+    async def handle_cli(
+        self,
+        files: Optional[str] = None,
+        question: Optional[str] = None,
+        chat: bool = False,
+        verbose: bool = False,
+        clear: bool = False,
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """
+        Entrypoint for local document RAG CLI tool.
+        """
+        if clear:
+            # delete self.persist_dir directory including all subdirectories and files
+            if os.path.exists(self.persist_dir):
+                # Ask for confirmation
+                response = input(
+                    f"Are you sure you want to delete data within {self.persist_dir}? [y/N] "
+                )
+                if response.strip().lower() != "y":
+                    print("Aborted.")
+                    return
+                os.system(f"rm -rf {self.persist_dir}")
+            print(f"Successfully cleared {self.persist_dir}")
+
+        self.verbose = verbose
+        ingestion_pipeline = cast(IngestionPipeline, self.ingestion_pipeline)
+        if self.verbose:
+            print("Saving/Loading from persist_dir: ", self.persist_dir)
+        if files is not None:
+            documents = []
+            for _file in iglob(files, recursive=True):
+                _file = os.path.abspath(_file)
+                if os.path.isdir(_file):
+                    reader = SimpleDirectoryReader(input_dir=_file, filename_as_id=True)
+                else:
+                    reader = SimpleDirectoryReader(
+                        input_files=[_file], filename_as_id=True
+                    )
+
+                documents.extend(reader.load_data(show_progress=verbose))
+
+            await ingestion_pipeline.arun(show_progress=verbose, documents=documents)
+            ingestion_pipeline.persist(persist_dir=self.persist_dir)
+            self.on_ingestion(documents)
+
+        if question is not None:
+            await self.handle_question(question)
+        if chat:
+            await self.start_chat_repl()
+
+    async def handle_question(self, question: str) -> None:
+        if self.query_pipeline is None:
+            raise ValueError("query_pipeline is not defined.")
+        query_pipeline = cast(QueryPipeline, self.query_pipeline)
+        query_pipeline.verbose = self.verbose
+        response = query_pipeline.run(query_str=question)
+
+        if isinstance(response, StreamingResponse):
+            response.print_response_stream()
+        else:
+            response = cast(Response, response)
+            print(response)
+
+    async def start_chat_repl(self) -> None:
+        """
+        Start a REPL for chatting with the agent.
```

Comment:
Just a note, this isn't really chat since there is no memory.

Reply:
Just saw we have [`QueryPipelineAgentWorker`](https://github.com/run-llama/llama_index/blob/5a0ae84909043693f4920fc37e2b09f3668bac40/llama_index/agent/custom/pipeline_worker.py#L51)! Will add that in. Although, looks like it may not support streaming?

Reply:
Oh this was literally just added yday haha

Reply:
FYI ended up using `CondenseQuestionChatEngine` instead